### PR TITLE
Fix gosec issue: Potential file inclusion via variable

### PR DESCRIPTION
**Pull request by ezrasilvera**
Signed-off-by: Ezra Silvera <ezra@il.ibm.com>

**What this PR does / why we need it**:
Analyze all cases of this static analysis warning and make sure a file injection attack is not possible or relevant 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
"NONE"
```

### CODE REVIEW

**Review by xpivarc — CHANGES_REQUESTED** [review]

File: pkg/virt-launcher/monitor.go [hunk 1]
```diff
@@ -191,6 +191,7 @@ func (mon *monitor) RunForever(startTimeout time.Duration, signalStopChan chan s
 	mon.monitorLoop(startTimeout, signalStopChan)
 }
 
+// #nosec doesn't seem like this function is even used
```
[the remark this answers is not in the document]
Comment (ezrasilvera) [reply pkg/virt-launcher/monitor.go]: Done. Removed it in separate commit.

File: pkg/certificates/triple/cert/cert.go [hunk 2]
```diff
@@ -161,8 +161,10 @@ func GenerateSelfSignedCertKeyWithFixtures(host string, alternateIPs []net.IP, a
 	certFixturePath := path.Join(fixtureDirectory, baseName+".crt")
 	keyFixturePath := path.Join(fixtureDirectory, baseName+".key")
 	if len(fixtureDirectory) > 0 {
+		// #nosec No risk for path injection. Used internally to read cert files
```
Comment (xpivarc) [inline pkg/certificates/triple/cert/cert.go:164]: I think this function is not used. Too be exact the function which calls this is not used anymore.

File: pkg/ephemeral-disk-utils/utils.go [hunk 3]
```diff
@@ -94,6 +94,7 @@ func FileExists(path string) (bool, error) {
 }
 func Md5CheckSum(filePath string) ([]byte, error) {
 
+	// #nosec no risk for path injection: only used to calculate MD5 of files
```
Comment (xpivarc) [inline pkg/ephemeral-disk-utils/utils.go:97]: I wasn't able to find a issue here. Do you think we can improve the Gosec rule?

File: pkg/util/net/ip/ip.go [hunk 4]
```diff
@@ -75,6 +75,7 @@ func NormalizeIPAddress(ipAddress string) string {
 }
 
 func isIPv6Disabled(filename string) bool {
+	// #nosec No injection path risk: filename is set to the const `disableIPv6Path`
```
Comment (xpivarc) [inline pkg/util/net/ip/ip.go:78]: Same here, I don't see how this can be changed.

**Review by xpivarc — CHANGES_REQUESTED** [review]
I found some dead code. Other than that I would probably prefer using `filepath.Clean` at earliest occurance of these variables. Maybe we can contribute back to Gosec and make this rule 'smarter'? WDYT?

**Review by dhiller — APPROVED** [review]
/lgtm
/approve
/hold in case a second reviewer wants to add anything

However, are you still planning to make the gosec rule smarter? I guess this will happen in a separate PR then?

**Comment by ezrasilvera** [comment]
> /lgtm
> /approve
> /hold in case a second reviewer wants to add anything
> 
> However, are you still planning to make the gosec rule smarter? I guess this will happen in a separate PR then?

Yes. Let's hold with this - I want to do another pass trying to improve some of the warnings

File: pkg/certificates/triple/cert/cert.go (hunk 2 above)
Comment (ezrasilvera) [reply pkg/certificates/triple/cert/cert.go]: I removed both functions

File: pkg/ephemeral-disk-utils/utils.go (hunk 3 above)
Comment (ezrasilvera) [reply pkg/ephemeral-disk-utils/utils.go]: The Open also need to be cleaned

**Pull request merged**